# Mini Project 2- Exploring AI Use by Demographic Features

## Using Dataset: Pew Research ATP W152, August 2024

In [33]:
import pandas as pd
import numpy as np

## Include ATP W152- August 2024 Dataset

In [34]:
data = pd.read_csv('data/August 2024/ATP W152.csv')

In [35]:
display(data.describe())
display(data.head())

,QKEY,DEVICE_TYPE_W152,SVYMODE_W152,LANG_W152,FORM_W152,AI_HEARD_W152,CNCEXC_W152,USEAI_W152,AICONTROL1_W152,AICONTROL2_W152,...,F_PARTYLN_FINAL,F_PARTYSUM_FINAL,F_PARTYSUMIDEO_FINAL,F_REG,F_INC_SDT1,F_IDEO,F_INTFREQ,F_VOLSUM,F_INC_TIER2,WEIGHT_W152
count,5.410000e+03,5410.000000,5410.000000,5410.000000,5410.000000,5410.000000,5410.00000,5410.000000,5410.000000,5410.000000,...,2365.000000,5410.000000,5410.000000,5410.000000,5410.000000,5410.000000,5410.000000,5410.000000,5410.000000,5410.000000
mean,1.869364e+11,1.688725,1.039741,9.023660,1.500739,1.913124,2.82902,4.211460,4.131423,2.363216,...,9.648626,1.797597,2.782810,1.412754,10.103327,4.836969,2.125508,2.201664,7.882625,1.000000
std,5.318192e+10,0.960277,0.195369,0.152001,0.500046,5.480991,7.33057,7.546541,7.155505,5.464507,...,26.944175,1.483587,1.794066,0.767634,20.203061,13.480087,6.403944,6.997570,23.273008,0.595664
min,1.003140e+05,1.000000,1.000000,9.000000,1.000000,1.000000,1.00000,1.000000,1.000000,1.000000,...,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,0.135260
25%,2.018011e+11,1.000000,1.000000,9.000000,1.000000,1.000000,2.00000,2.000000,3.000000,2.000000,...,1.000000,1.000000,1.000000,1.000000,3.000000,2.000000,1.000000,1.000000,1.000000,0.685758
50%,2.020010e+11,1.000000,1.000000,9.000000,2.000000,2.000000,2.00000,4.000000,4.000000,2.000000,...,2.000000,2.000000,3.000000,1.000000,6.000000,3.000000,2.000000,2.000000,2.000000,0.907527
75%,2.024000e+11,3.000000,1.000000,9.000000,2.000000,2.000000,3.00000,5.000000,4.000000,2.000000,...,2.000000,2.000000,4.000000,1.000000,9.000000,4.000000,2.000000,2.000000,3.000000,1.161045
max,2.024000e+11,4.000000,2.000000,10.000000,2.000000,99.000000,99.00000,99.000000,99.000000,99.000000,...,99.000000,9.000000,9.000000,3.000000,99.000000,99.000000,99.000000,99.000000,99.000000,3.737946


,QKEY,INTERVIEW_START_W152,INTERVIEW_END_W152,DEVICE_TYPE_W152,SVYMODE_W152,LANG_W152,FORM_W152,AI_HEARD_W152,CNCEXC_W152,USEAI_W152,...,F_PARTYLN_FINAL,F_PARTYSUM_FINAL,F_PARTYSUMIDEO_FINAL,F_REG,F_INC_SDT1,F_IDEO,F_INTFREQ,F_VOLSUM,F_INC_TIER2,WEIGHT_W152
0,100314,15-Aug-2024 15:30:49,15-Aug-2024 15:40:52,1,1,9,1,2,2,5,...,NaN,1,2,1,7,3,3,1,2,0.639769
1,103379,15-Aug-2024 13:58:04,15-Aug-2024 14:21:23,3,1,9,1,2,2,5,...,NaN,1,1,1,9,2,5,2,3,1.338840
2,104727,15-Aug-2024 07:18:22,15-Aug-2024 07:24:47,1,1,9,2,2,3,5,...,NaN,1,1,1,99,2,2,2,99,0.909360
3,106590,13-Aug-2024 22:30:51,13-Aug-2024 22:36:57,1,1,9,1,2,2,5,...,1.0,1,2,1,8,3,2,1,2,0.919896
4,107668,13-Aug-2024 15:23:41,13-Aug-2024 15:30:23,1,1,9,2,2,2,3,...,1.0,1,2,1,9,3,1,1,3,0.719993


In [36]:
for i in data.columns.to_list():
    print(i)

QKEY
INTERVIEW_START_W152
INTERVIEW_END_W152
DEVICE_TYPE_W152
SVYMODE_W152
LANG_W152
FORM_W152
AI_HEARD_W152
CNCEXC_W152
USEAI_W152
AICONTROL1_W152
AICONTROL2_W152
AICHANGE_W152
AIFUTRIMPCT_a_W152
AIFUTRIMPCT_b_W152
AIFUTRIMPCT_c_W152
AIFUTRIMPCT_d_W152
AIFUTRIMPCT_e_W152
AIFUTRIMPCT_f_W152
AIFUTRIMPCT_g_W152
AIFUTRIMPCT_h_W152
AIFUTRIMPCT_i_W152
AIFUTRIMPCT_j_W152
AIJOBS_W152
AIJOBIMPCT_a_W152
AIJOBIMPCT_b_W152
AIJOBIMPCT_c_W152
AIJOBIMPCT_d_W152
AIJOBIMPCT_e_W152
AIJOBIMPCT_f_W152
AIJOBIMPCT_g_W152
AIJOBIMPCT_h_W152
AIJOBIMPCT_i_W152
AIJOBIMPCT_j_W152
HUMANVAI_a_W152
HUMANVAI_b_W152
HUMANVAI_c_W152
HUMANVAI_d_W152
HUMANVAI_e_W152
HUMANVAI_f_W152
HUMANVAI_g_W152
HUMANVAI_h_W152
PERSBENHRM_W152
TRSTAIPRS_W152
AICONCERN_a_W152
AICONCERN_b_W152
AICONCERN_c_W152
AICONCERN_d_W152
AICONCERN_e_W152
AICONCERN_f_W152
AICONCERN_g_W152
FUTRAI_a_W152
FUTRAI_b_W152
FUTRAI_c_W152
FUTRAI_d_W152
DISCRIM1_a_W152
DISCRIM1_b_W152
DISCRIM1_c_W152
DISCRIM1_d_W152
DISCRIM1_e_W152
DISCRIM1_f_W152
AIREG_W152

In [37]:
data.drop_duplicates(inplace=True)
data.dropna(how='all', inplace=True)

## Subset for Regression Models
- `QKEY` is the unique identifier (id) per row
- Including predictor and outcome variables planned for logit and ols models
- Outcome variables: 
    - `CHATUSE_W152` is binary (coded 1, 2) for logit
    - `USEAI_W152` is likert for ols

In [38]:
predictor_cols = ['F_INC_SDT1', 'F_EDUCCAT2', 'F_AGECAT', 'F_GENDER', 'F_RACECMB', 'F_RACETHNMOD']
outcome_cols = ['CHATUSE_W152', 'USEAI_W152']
include_cols = ['QKEY'] + predictor_cols + outcome_cols
data = data[include_cols]
data.head(20)

,QKEY,F_INC_SDT1,F_EDUCCAT2,F_AGECAT,F_GENDER,F_RACECMB,F_RACETHNMOD,CHATUSE_W152,USEAI_W152
0,100314,7,3,4,1,1,1,NaN,5
1,103379,9,3,4,1,1,1,NaN,5
2,104727,99,2,3,1,1,1,NaN,5
3,106590,8,6,3,2,1,1,NaN,5
4,107668,9,5,3,2,1,1,2.0,3
5,108435,9,6,4,2,1,1,2.0,2
6,108592,5,4,4,2,2,2,2.0,3
7,112238,6,2,4,2,1,1,NaN,5
8,112371,3,4,3,2,1,3,2.0,5
9,116422,5,2,4,2,1,1,NaN,5


## Recode Variables

In [39]:
# 99 = "Refused" code; recode as missing
recode_99_cols = predictor_cols + outcome_cols
for col in recode_99_cols:
    data[col] = data[col].replace(99, np.nan)

# Remap Binary Outcome Variable
data['CHATUSE_W152'] = data['CHATUSE_W152'].replace({2: 0})  # 1 = Yes, have used, 0 = No, have not used

display(data.describe())
display(data.head(20))

,QKEY,F_INC_SDT1,F_EDUCCAT2,F_AGECAT,F_GENDER,F_RACECMB,F_RACETHNMOD,CHATUSE_W152,USEAI_W152
count,5.410000e+03,5150.000000,5397.000000,5383.000000,5387.000000,5327.000000,5364.000000,4027.000000,5377.000000
mean,1.869364e+11,5.615340,3.821197,2.625859,1.506590,1.631312,1.960291,0.497393,3.629719
std,5.318192e+10,3.095709,1.564434,1.017907,0.511385,1.091040,1.348927,0.500055,1.344893
min,1.003140e+05,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,0.000000,1.000000
25%,2.018011e+11,3.000000,2.000000,2.000000,1.000000,1.000000,1.000000,0.000000,2.000000
50%,2.020010e+11,6.000000,4.000000,3.000000,2.000000,1.000000,1.000000,0.000000,4.000000
75%,2.024000e+11,9.000000,5.000000,4.000000,2.000000,2.000000,3.000000,1.000000,5.000000
max,2.024000e+11,9.000000,6.000000,4.000000,3.000000,5.000000,5.000000,1.000000,5.000000


,QKEY,F_INC_SDT1,F_EDUCCAT2,F_AGECAT,F_GENDER,F_RACECMB,F_RACETHNMOD,CHATUSE_W152,USEAI_W152
0,100314,7.0,3.0,4.0,1.0,1.0,1.0,NaN,5.0
1,103379,9.0,3.0,4.0,1.0,1.0,1.0,NaN,5.0
2,104727,NaN,2.0,3.0,1.0,1.0,1.0,NaN,5.0
3,106590,8.0,6.0,3.0,2.0,1.0,1.0,NaN,5.0
4,107668,9.0,5.0,3.0,2.0,1.0,1.0,0.0,3.0
5,108435,9.0,6.0,4.0,2.0,1.0,1.0,0.0,2.0
6,108592,5.0,4.0,4.0,2.0,2.0,2.0,0.0,3.0
7,112238,6.0,2.0,4.0,2.0,1.0,1.0,NaN,5.0
8,112371,3.0,4.0,3.0,2.0,1.0,3.0,0.0,5.0
9,116422,5.0,2.0,4.0,2.0,1.0,1.0,NaN,5.0


In [40]:
# Save cleaned/recoded data for regression analysis
data.to_csv('data/August 2024/atp-w152-clean.csv', index=False)

## Logistic Regression

In [41]:
data = data.dropna()
data.describe()

,QKEY,F_INC_SDT1,F_EDUCCAT2,F_AGECAT,F_GENDER,F_RACECMB,F_RACETHNMOD,CHATUSE_W152,USEAI_W152
count,3.796000e+03,3796.000000,3796.000000,3796.000000,3796.000000,3796.000000,3796.000000,3796.000000,3796.000000
mean,1.878028e+11,6.089305,4.089831,2.452845,1.463119,1.667808,2.043730,0.500263,3.382771
std,5.176870e+10,2.997791,1.517085,0.979979,0.513798,1.101216,1.427265,0.500066,1.337296
min,1.076680e+05,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,0.000000,1.000000
25%,2.018011e+11,3.000000,3.000000,2.000000,1.000000,1.000000,1.000000,0.000000,2.000000
50%,2.020010e+11,7.000000,5.000000,2.000000,1.000000,1.000000,1.000000,1.000000,3.000000
75%,2.024000e+11,9.000000,5.000000,3.000000,2.000000,2.000000,3.000000,1.000000,5.000000
max,2.024000e+11,9.000000,6.000000,4.000000,3.000000,5.000000,5.000000,1.000000,5.000000


In [42]:
import statsmodels.api as sm
import statsmodels.formula.api as smf

np.random.seed(42)

# Gender_2 Female
# Gender_3 In some other way
gender_dummies = pd.get_dummies(data['F_GENDER'], prefix='Gender', drop_first=True, dtype=int)
race_dummies = pd.get_dummies(data['F_RACECMB'].astype(str), prefix='Race', drop_first=True, dtype=int)

X = pd.concat([data[['F_INC_SDT1','F_EDUCCAT2','F_AGECAT']], gender_dummies, race_dummies], axis=1)
y = data["CHATUSE_W152"]

X = sm.add_constant(X)

# Renaming columns for interpretability
rename_dict = {
    "const": "Intercept",
    "F_INC_SDT1": "Income",
    "F_AGECAT": "Age",
    "F_EDUCCAT2": "Education",
    "Gender_2.0": "Gender: Female",
    "Gender_3.0": "Gender: In some other way",
    "Race_2.0": "Race: Black or African-American",
    "Race_3.0": "Race: Asian or Asian-American",
    "Race_4.0": "Race: Mixed Race",
    "Race_5.0": "Race: Other"  
}

X = X.rename(columns=rename_dict)

logreg = sm.Logit(y, X)
logreg_results = logreg.fit()

Optimization terminated successfully.
         Current function value: 0.615824
         Iterations 5


In [43]:
print(logreg_results.summary()) # the regular table is not very intepretable because of log odds

                           Logit Regression Results                           
Dep. Variable:           CHATUSE_W152   No. Observations:                 3796
Model:                          Logit   Df Residuals:                     3786
Method:                           MLE   Df Model:                            9
Date:                Wed, 20 May 2026   Pseudo R-squ.:                  0.1116
Time:                        22:15:03   Log-Likelihood:                -2337.7
converged:                       True   LL-Null:                       -2631.2
Covariance Type:            nonrobust   LLR p-value:                1.266e-120
                                      coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------------------
Intercept                           0.2817      0.145      1.947      0.052      -0.002       0.565
Income                              0.0722      0.013      5.435    

In [44]:
df_odds = logreg_results.conf_int()
df_odds["Odds Ratio"] = logreg_results.params
df_odds.columns = ["CI lower", "CI Upper", "Odds Ratio"]

# converting from log odds to odds ratios for intepretabiliy
df_odds = np.exp(df_odds)

# percent change
df_odds["% Change"] = (df_odds["Odds Ratio"] - 1) * 100

# CI effect
conditions = [
    (df_odds["CI lower"] > 1.0) & (df_odds["CI Upper"] > 1.0),  # Entirely above 1
    (df_odds["CI lower"] < 1.0) & (df_odds["CI Upper"] < 1.0),  # Entirely below 1
]
choices = ["Positive Effect", "Negative Effect"]
df_odds["CI Effect"] = np.select(conditions, choices, default="No Effect")

# P-value
df_odds["P-value"] = logreg_results.pvalues
df_odds["Significant"] = df_odds["P-value"] < 0.05

df_odds = df_odds[["Odds Ratio","% Change","CI lower","CI Upper","CI Effect","P-value","Significant"]]

df_odds

,Odds Ratio,% Change,CI lower,CI Upper,CI Effect,P-value,Significant
Intercept,1.325429,32.542878,0.998145,1.760027,No Effect,5.151933e-02,False
Income,1.074882,7.488221,1.047252,1.103242,Positive Effect,5.487784e-08,True
Education,1.272070,27.206990,1.208954,1.338481,Positive Effect,1.893266e-20,True
Age,0.512486,-48.751379,0.475278,0.552608,Negative Effect,1.118792e-67,True
Gender: Female,0.696018,-30.398188,0.604240,0.801736,Negative Effect,5.090978e-07,True
Gender: In some other way,1.063818,6.381795,0.470161,2.407064,No Effect,8.819530e-01,False
Race: Black or African-American,1.217225,21.722474,0.976063,1.517972,No Effect,8.100208e-02,False
Race: Asian or Asian-American,1.630574,63.057429,1.311896,2.026664,Positive Effect,1.049278e-05,True
Race: Mixed Race,1.059757,5.975746,0.772298,1.454213,No Effect,7.192170e-01,False
Race: Other,0.957151,-4.284864,0.660079,1.387923,No Effect,8.173273e-01,False


## Interpretation

### Income
As CI bounds are above 1.0, income has a positive effect on AI usage. The result is significant (p < 0.05). With one unit increase in income, the odds of using AI increase by 7.48%. 

### Education
As CI bounds are above 1.0, education has a positive effect on AI usage. The result is significant (p < 0.05). With one unit increase in education, the odds of using AI increase by 27.20%. 

### Age
As CI bounds are below 1.0, age has a negative effect on AI usage. The result is significant (p < 0.05). For each one bracket increase in the age category (moving up from 18–29, to 30–49, to 50–64, and 65+), the odds of using AI decrease by 48.75%. 

### Gender: Female [baseline Male]
As CI bounds are below 1.0, female gender has a negative effect on AI usage. The result is significant (p < 0.05). For females the odds of using AI decrease by 30.39% compared to males. 

### Gender: In some other way [baseline Male]
As CI bounds include 1.0, there is no significant effect of "in some other way" gender on AI usage. The result is not significant (p > 0.05). 

### Race: Black or African-American [baseline White]
As CI bounds include 1.0, there is no significant effect of Black or African-American Race on AI usage. The result is not significant (p > 0.05). 

### Race: Asian or Asian-American [baseline White]
As CI bounds are above 1.0, Asian or Asian-American Race has a positive effect on AI usage. For Asian or Asian-American Race the odds of using AI increase by 63.05% compared to White. 

### Race: Mixed Race [baseline White]
As CI bounds include 1.0, there is no significant effect of Mixed Race on AI usage. The result is not significant (p > 0.05). 

### Race: Other [baseline White]
As CI bounds include 1.0, there is no significant effect of Black or Other Race on AI usage. The result is not significant (p > 0.05). 